deform/linear_deformation.py

```diff
@@ -0,0 +1,1173 @@
+            must have the same number of dimensions as ``par_space``.
+        """
+        # The operator maps from the parameter space to the template
+        # (image) space.
+        self.template = template
+        self.domain_space = odl.ProductSpace(self.template.space,
+                                             self.template.space.ndim)
+
+        super().__init__(self.domain_space, self.template.space, linear=False)
+
+    def _call(self, displacement):
+        """Implementation of ``self(displacement)``.
+
+        Parameters
+        ----------
+        displacement: `ProductSpaceVector`
+            Linearized deformation parameters for image grid points.
+        """
+        image_pts = self.template.space.grid.points()
+        image_pts += np.asarray(displacement).T
+
+        return self.template.interpolation(image_pts.T, bounds_check=False)
+
+    def linear_deform(self, template, displacement):
+        """Implementation of ``self(template, displacement)``.
+
+        Parameters
+        ----------
+        template : `DiscreteLpVector`
+            Fixed template deformed by the vector field. Its space
+            must have the same number of dimensions as ``par_space``.
+        displacement: `ProductSpaceVector`
+            Linearized deformation parameters for image grid points.
+        """
+        image_pts = template.space.grid.points()
+        image_pts += np.asarray(displacement).T
+        return template.interpolation(image_pts.T, bounds_check=False)
+
+    def derivative(self, displacement):
+        """Frechet derivative of this operator in ``disp``.
+
+        Parameters
+        ----------
+        displacement: `ProductSpaceVector`
+            Linearized deformation parameters for image grid points.
+
+        Returns
+        -------
+        deriv : `Operator`
+        The derivative of this operator, evaluated at ``displacement``
+        """
+        deriv_op = LinearizedDeformationDerivative(self.template,
+                                                   displacement)
+        return deriv_op
+
+
+class LinearizedDeformationDerivative(LinearizedDeformationOperator):
+
+    """Frechet derivative of the Linearized deformation operator."""
+
+    def __init__(self, template, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        displacement: `ProductSpaceVector`
+            Linearized deformation parameters for image grid points.
+        template : `DiscreteLpVector`
+            Fixed template deformed by the vector field. Its space
+            must have the same number of dimensions as ``par_space``.
+        """
+
+        super().__init__(template)
+
+        self.template = template
+        self.displacement = displacement
+
+        Operator.__init__(self, self.displacement.space, self.template.space,
+                          linear=True)
+
+    def derivative(self, *args, **kwargs):
```

Comment:
No need to add a `NotImplementedError` for `derivative`, a deafult already exists in `Operator`